gateway-ha/src/test/java/com/lyft/data/gateway/ha/router/TestPrestoQueueLengthRoutingTable.java

```diff
@@ -197,6 +236,70 @@ public void testRoutingWithSkewedWeightDistribution() {
     }
   }
 
+  @Test
+  public void testRoutingWithUserQueuedLength() {
+    int numBackends = 2;
+    int queryVolume = 10000;
+
+    // Case 1: All user queue counts Present.
+    // Validate always routed to  cluster with lowest user queue
+    registerBackEndsWithUserQueue(mockRoutingGroup, numBackends, Arrays.asList(1, 2));
+    for (int i = 0; i < queryVolume; i++) {
+      assertEquals(routingTable.getEligibleBackEnd(mockRoutingGroup, mockUser),
+              mockRoutingGroup + "0");
+    }
+
+    // Case 2: Not all user queue counts Present.
+    // Validate always routed to cluster with zero queue length i.e. the missing cluster.
+    registerBackEndsWithUserQueue(mockRoutingGroup, numBackends, Arrays.asList(1));
```

Comment:
It will be missing as the API we use to get queued queries wont' return that user so if a cluster has zero queue length for the user.

This here is a helper method to create the  {user -> {cluster -> queue}} map and mimic that behavior i.e. zero queue lengths will be missing in the map .